### PR TITLE
Add breakpoint based modal variations

### LINKED ISSUE #522: Modal versions for breakpoints
## Description
As proposed in a discussion and requested by the team, we need to add three versions of the Modal component for all breakpoints we are currently using. The Modal versions are the followings, copied from the proposal discussion:

### Modal S
A flexible full-size page-like with 8-cols, `16px` gutter, and `24px` left and right paddings. The close action is top-left aligned with a `margin-top`, `margin-left`, and `margin-bottom` of `8px`.

### Modal M
A `768px` fixed width with 8-cols, `16px` gutter, and `64px` left and right paddings. The close action is placed outside and top-right aligned with a `margin-top` and `margin-bottom` of `8px`.

### Modal L
A `1024px` fixed width with 8-cols, `32px` gutter, and `64px` left and right paddings. The close action is placed outside and top-right aligned with a `margin-top` of `16px` and `margin-bottom` of `8px`.

## References
The Modals demanded a new component: the **Close modal** button. All design documentation can be found here:
1. **Initial discussion: #507**
2. **[Modal component in Figma, Views page](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/?node-id=344%3A285)**
3. **[Close modal button component in Figma, Components page](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?node-id=1320%3A3077)**

The component is related to issue #458

### LINKED ISSUE #1375: Openverse logo shifts when opening filter sidebar
## Description

When opening the filter sidebar on mobile in the results page, the Openverse logo shifts up and down slightly.

## Reproduction

1. Visit https://search-staging.openverse.engineering/search/image?q=food on mobile
2. Open the filter sidebar
3. Observe that the logo shifts slightly

## Screenshots

https://cln.sh/qLyzpg

## Environment

 - Device: Mobile
 - OS: Android & iPhone
 - Browser: Chrome & Firefox

## Additional context

From @panchovm

> We fixed this bug for one of the two menus but now it is broken on both

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #522 by @panchovm 
Fixes #1375 by @AetherUnbound 

## Description

Updates the existing modal implementation to add breakpoint-based variations. It's just style updates except that the mobile version also includes the logo button. I added that because the only place we actually currently use the modal is for the mobile filters and search type switcher and those use the logo button in the header.

This does also wrap the top bar in a slot to make it easier to modify the modal content without having to copy/paste the whole thing like we've done with the `VMobileModalContent`.

This PR also removes that component :tada: 

## Testing Instructions

Checkout the branch and run `pnpm dev` to visit the site locally.

Go to the search results page with a small viewport width and verify that the modal for the search type switcher and the filters still works as expected and that it looks good.

Next run `pnpm storybook` and open the Modal story in storybook and test out the various breakpoints using Storybook's breakpoints option or by opening the iframe popout and changing your viewport width manually (the second option can give you a sense of how responsive it is).

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by panchovm** [comment]
It looks great so far. I just noticed two things that maybe you are currently working on.

## Modal scroll
When I scroll down, the white area where content is placed remains fixed while the internal content moves up and down. I think the best approach is how Unsplash and Rawpixel solved this. Go and search for something since pasting a URL from both sites takes you to the non-modal version of the single result view.

## Modal width
The modal widths for breakpoints `2xl` and `xl` work well, but while changing the viewport manually, I notice the width changes as the page view increases or decreases. I think we should keep the modal width fixed until reaching a different breakpoint. In that way, the content placed inside does not move too much and the modal is always center aligned.

In breakpoints `lg` and `md`, the modal width is smaller than the [Design Libary mockup](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?node-id=1327%3A4348). For `lg` it should be `768px` while for `md` should be full width (`768px`). Below two screenshots of what I am seeing.

### Screenshots
|`lg`|`md`|
|:---|:---|
|CleanShot 2022-04-14 at 16 13 26@2x|CleanShot 2022-04-14 at 16 13 10@2x|

**Comment by sarayourfriend** [comment]
Okay, I didn't understand the mock-ups correctly, I will update it, it's just a matter of moving the widths up one breakpoint.

> When I scroll down, the white area where content is placed remains fixed while the internal content moves up and down. I think the best approach is how Unsplash and Rawpixel solved this. Go and search for something since pasting a URL from both sites takes you to the non-modal version of the single result view.

How would we implement that behavior with the close button where it is, above the modal? It would scroll out of view :thinking:

**Comment by panchovm** [comment]
I do not think that being out of view when scrolling down is terrible as it is consistent with all breakpoints and the back to results. On the other hand, the escape button can work for closing the modal.

**Review by obulat — COMMENTED** [review]
I really like the clean-up of the Modal components :)

**Comment by sarayourfriend** [comment]
@panchovm widths are updated, let me know if they're correct now (I'm having a hard time fully understanding the mock-ups and how they relate to the space between the breakpoints, in particular the space between the `sm` and `md` breakpoints. I took a guess, but let me know if it makes sense.

I also updated it so that it scrolls the full modal overlay rather than just the modal body, as you requested.

**Comment by panchovm** [comment]
### Modal behavior

I run dev and it does not look like a modal but like page that keeps the openverse header fixed. Here is a screenshot.

|Non-scroll|Scroll|
|:---|:---|
|CleanShot 2022-04-25 at 11 23 00@2x|CleanShot 2022-04-25 at 11 24 05@2x|

---

### Close area

In `md` I am seeing the close action outside the modal, whereas in `sm` and `xs` the close area has the same close area as the content switcher/filters content. The logic you are following is fine as it is consistent across all content displayed as full-size, but the design proposes a "back to results" top-left button.

I am considering requesting an update of `Close modal` component to have a close area for modals, content switcher and filters, and header scrolled version. Here is a screenshot of how it might look like.

CleanShot 2022-04-25 at 11 32 13@2x

Let me know what you think

**Comment by sarayourfriend** [comment]
@panchovm Sorry, there is some confusion. This does not implement the "results open as a modal" behavior, it just updates the modal to behave appropriately for each breakpoint (a necessary step for the results as a modal PR).

When we implement the "results as a modal" PR, we will replace the top bar with the "back to search results" as indicated in the designs. This is easily possible now due to some internal implementation changes I made in this PR as well.

Basically this PR sets us up for implementing the search result modal (so long as we can sort out the browser navigation difficulties, but that's a technical implementation detail, not something you need to worry about :grin:)

**Comment by panchovm** [comment]
Perfect. Thanks for the clarification. The dev work looks great then ✨

**Review by obulat — APPROVED** [review]
Looks nice, and the code is much cleaner.
You cannot open a modal twice, but I guess it's being worked on in #1279?

**Review by zackkrida — APPROVED** [review]
These look great, and such a clean approach to fixing the issue. I still see a vertical 1px (!!) shift in the logo when opening the menu, which I tried to debug but was unsuccessful. It's a major improvement regardless.
